test/e2e/useroauthaccesstokens_test.go

```diff
@@ -0,0 +1,560 @@
+		},
+	}
+
+	for _, token := range tokens {
+		_, err := oauthClient.OAuthAccessTokens().Create(testCtx, token, metav1.CreateOptions{})
+		if err != nil {
+			errors = append(errors, err)
+		}
+		trashBin.AddResource(oauthv1.GroupVersion.WithResource("oauthaccesstokens"), token.GetObjectMeta())
+	}
+
+	return
+}
+
+type userOAuthAccessTokenTestFunc func(t *testing.T, adminConfig *rest.Config, testNS, user1uid, user2uid, user1token, user2token string)
+
+func testUserOAuthAccessAPI(t *testing.T, test userOAuthAccessTokenTestFunc) {
+	adminConfig := NewClientConfigForTest(t)
+	testTrashBin := NewResourceTrashbin(t, adminConfig)
+	defer testTrashBin.Empty(t)
+
+	kubeClient, err := kubernetes.NewForConfig(adminConfig)
+	require.NoError(t, err)
+	projectClient, err := projectclient.NewForConfig(adminConfig)
+	require.NoError(t, err)
+
+	testNS := CreateTestProject(t, kubeClient, projectClient)
+	testTrashBin.AddResource(corev1.SchemeGroupVersion.WithResource("namespaces"), testNS.GetObjectMeta())
+
+	frantaUID, mirkaUID, frantaTokenString, mirkaTokenString, trashBin, errors := setupTokenStorage(t, adminConfig, testNS.Name)
+	require.Equal(t, len(errors), 0)
+	testTrashBin.Merge(trashBin)
+
+	test(t, adminConfig, testNS.Name, frantaUID, mirkaUID, frantaTokenString, mirkaTokenString)
+}
+
+func TestUserOAuthAccessTokensList(t *testing.T) {
+	testUserOAuthAccessAPI(t, listTokens)
+}
+
+func TestUserOAuthAccessTokensGet(t *testing.T) {
+	testUserOAuthAccessAPI(t, getTokens)
+}
+
+func TestUserOAuthAccessTokensDelete(t *testing.T) {
+	testUserOAuthAccessAPI(t, deleteTokens)
+}
+
+func TestUserOAuthAccessTokensWatch(t *testing.T) {
+	testUserOAuthAccessAPI(t, watchTokens)
+}
+
+func listTokens(t *testing.T, adminConfig *rest.Config, testNS, _, _, frantaTokenString, mirkaTokenString string) {
+	testCtx := context.Background()
+
+	tests := []struct {
+		name            string
+		userToken       string
+		userName        string
+		fieldSelector   fields.Selector
+		labelSelector   labels.Selector
+		expectedResults int
+		expectedError   string
+	}{
+		{
+			name:            "happy path",
+			userToken:       mirkaTokenString,
+			userName:        "mirka",
+			expectedResults: 2,
+		},
+		{
+			name:          "invalid field selector taken from oauthaccesstokens to match another username",
+			userToken:     frantaTokenString,
+			fieldSelector: fields.OneTermEqualSelector("userName", "pepa"),
+			expectedError: "is not a known field selector",
+		},
+		{
+			name:            "single-equal field selector to get own tokens by client",
+			userToken:       frantaTokenString,
+			userName:        "franta",
+			fieldSelector:   fields.OneTermEqualSelector("clientName", "openshift-browser-client"),
+			expectedResults: 2,
+		},
+		{
+			name:            "set label selector to get own tokens and of others",
+			userToken:       mirkaTokenString,
+			userName:        "mirka",
+			labelSelector:   parseLabelSelectorOrDie("randomLabel notin (mirka,franta)"),
```

Comment:
would add the simple case too: userName == somebody else.

Reply:
there's a fieldSelector like this, but I'll add a label to one of the basic tokens above and will attempt to list it